scripts/validate.mjs

```diff
@@ -64,3 +100,34 @@ function forEachFile(directoryPath, callback) {
 function readJson(filePath) {
   return JSON.parse(readFileSync(filePath, 'utf8'))
 }
+
+/**
+ * Iterates over each properties of objects specified in the provided JSON schema.
+ */
+function forEachObjectProperty(schema, callback) {
+  if (Array.isArray(schema)) {
```

Comment:
**suggestion:**
you could early return here if `schema` is `null`, this will simplify a little bit the rest of the code

Reply:
I don't see how it would simplify the code?

The `schema !== null` is directly related to `typeof schema === 'object'` (since `typeof null === 'object'`). By hoisting this condition, you would need to remember that `schema` cannot be `null` when coming across the `typeof schema === 'object'` because of the early exit.

I think keeping those two conditions together makes sense here.

Reply:
it is really a nitpick, nothing blocking. I get what you're doing in this part of the code, I just thought having something like this could simplify if (basically to simplify the if/else):

```
if (schema === null){
  return;
}
if (Array.isArray(schema)) {
  ....
}
else if (typeof schema === 'object') {

}
```